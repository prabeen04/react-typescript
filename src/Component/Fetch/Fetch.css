.user-wrapper{
    background: #0f0f0f;
   height: 100vh;
}
.user-list{
    flex-basis: 25%;
}
.user-detail{
    flex-basis: 70%;
    height: 500px;
    margin: 0.5rem;
    padding: 1rem;
    border: 1px solid rgb(245, 111, 1);
}
.fetching{
    color: rgb(245, 111, 1);
    text-align: center;
}
.user-list-item{
    padding: 0.5rem;
    border: 1px solid rgb(56, 44, 29);
    border-radius: 0.3rem;
    margin: 0.6em;
    color: rgb(245, 111, 1);
    font-family: Margarine;
    cursor: pointer;
    display: flex;
    flex-wrap: wrap
}
.user-list-item:hover{
    color: rgb(245, 50, 1);
    border: none;
    box-shadow: 4px 4px 18px -6px  rgba(245, 90, 1, 0.829);
}
.active-list{
    color: rgb(245, 50, 1);
    box-shadow: 4px 4px 18px -6px  rgb(245, 50, 1);
}