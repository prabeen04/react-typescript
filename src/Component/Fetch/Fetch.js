import * as React from 'react'
import useUserData, { IUser } from "../../Hooks/UserData";
export default function Fetch() {
    const { fetchingUsers, fetchingUsersError, users: IUser } = useUserData();
    return (
        <>

        </>
    )
}
