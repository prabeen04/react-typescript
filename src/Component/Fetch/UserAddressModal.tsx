import * as React from 'react';
import { Modal, ModalHeader, ModalBody } from 'reactstrap';

export interface IUserAddressModal {
    visible: boolean;
    toggle: () => void;
    children: JSX.Element
}
function UserAddressModal(props: IUserAddressModal) {
    function toggle() { }
    return (
        <div>
            <Modal isOpen={true} toggle={toggle}>
                <ModalHeader toggle={toggle}>Modal title</ModalHeader>
                <ModalBody>{props.children}</ModalBody>
            </Modal>
        </div>
    );
}

export default UserAddressModal;