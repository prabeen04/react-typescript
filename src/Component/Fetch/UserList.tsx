import * as React from 'react';
import { IUser } from "../../Hooks/UserData";

export default function UserList(props: any) {
    const { activeUser, ...user } = props
    console.log(props)
    return (
        <div className={`user-list-item ${user.id === activeUser.id ? 'active-list' : ''}`}>
            {props.name}
        </div>
    )
}
