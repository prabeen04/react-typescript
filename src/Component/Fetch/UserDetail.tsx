import * as React from 'react';
import { useSpring, animated } from 'react-spring';
import { IUser, IAddress, IGeo, ICompany } from "../../Hooks/UserData";
import UserAddressModal from "../../Utils/UserAddressModal";
export interface IUserDetail {
    user: IUser;
}
export default function UserDetail(props: IUserDetail) {
    const { user } = props;
    const styles = useSpring({ opacity: 1, transform: 'translate3d(0px,0,0) ', from: { opacity: 0, transform: 'translate3d(0,200px,0) ', } });
    const [mapModal, setMapModal] = React.useState<boolean>(false)
    function toggleModal() {
        setMapModal(!mapModal)
    }
    return (
        <>
            <div className='user-detail'>
                <animated.div style={styles}>
                    <div className="user-detail-header">
                        <h1>
                            <i
                                className="fas fa-map-marker-alt clickable-icon"
                                onClick={toggleModal}
                            > &nbsp;&nbsp;</i>
                            {user.name} </h1>
                    </div>

                    <div className="user-detail-body">
                        <div className="detail-list-row">
                            <label>Email</label>
                            <p>{user.email}</p>
                        </div>
                        <div className="detail-list-row">
                            <label>Phone</label>
                            <p>{user.phone}</p>
                        </div>
                        <div className="detail-list-row">
                            <label>Website</label>
                            <a href={user.website} target='_blank'>{user.website}</a>
                        </div>
                    </div>
                    <div className="user-detail-body">
                        <div className="detail-list-row">
                            <label>Suite</label>
                            <p>{user.address.suite}</p>
                        </div>
                        <div className="detail-list-row">
                            <label>Street</label>
                            <p>{user.address.street}</p>
                        </div>
                        <div className="detail-list-row">
                            <label>City</label>
                            <p>{user.address.city}</p>
                        </div>
                        <div className="detail-list-row">
                            <label>Zipcode</label>
                            <p>{user.address.zipcode}</p>
                        </div>
                    </div>
                </animated.div>
            </div>
            <UserAddressModal
                noHeader
                visible={mapModal}
                toggle={toggleModal}
            >
                <p>{user.name}</p>
            </UserAddressModal>
        </>
    )
}