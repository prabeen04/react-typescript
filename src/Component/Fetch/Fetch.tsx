import * as React from 'react'
import useUserData, { IUserState, IUser, IAddress, ICompany, IGeo } from "../../Hooks/UserData";
import UserList from "./UserList";
import UserDetail from './UserDetail';
import './Fetch.css';

export default function Fetch() {
    const { fetchingUsers, fetchingUsersError, users } = useUserData();
    const [activeUser, setActiveUser] = React.useState<IUser>(users.length && users[0] || {})
    React.useEffect(() => {
        if (users.length) {
            setActiveUser(users[0])
        }
    }, [users])
    function setUser(user: IUser) {
        console.log(user)
        setActiveUser(user)
    }
    return (
        <>
            <div className='user-wrapper flex-container space-between'>
                {
                    fetchingUsers
                        ? <p className="fetching">fetching users...</p>
                        : fetchingUsersError
                            ? <p>Opps, Error while fetching users...</p>
                            : <>
                                <div className='user-list'>
                                    {users && users.map((user: IUser) => <UserList setUser={setUser} activeUser={activeUser} key={user.id} user={user} />)}
                                </div>
                                {activeUser.hasOwnProperty('name')
                                    && <UserDetail user={activeUser} />}
                            </>
                }
            </div>
        </>
    )
}
