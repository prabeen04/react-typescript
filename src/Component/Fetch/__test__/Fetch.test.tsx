import * as React from 'react';
import { shallow } from "enzyme";
import Fetch from '../Fetch';

describe('<Fetch/>', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<Fetch/>);
        expect(wrapper).toMatchSnapshot()
    })
})