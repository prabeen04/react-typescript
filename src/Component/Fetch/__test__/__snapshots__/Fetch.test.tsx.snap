// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fetch/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Fetch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="user-wrapper flex-container space-between"
      >
        <div
          className="user-detail-content"
        >
          <div
            className="user-list"
          />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="user-detail-content"
        >
          <div
            className="user-list"
          />
        </div>,
        "className": "user-wrapper flex-container space-between",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user-list"
            />,
            false,
          ],
          "className": "user-detail-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "user-list",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="user-wrapper flex-container space-between"
        >
          <div
            className="user-detail-content"
          >
            <div
              className="user-list"
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="user-detail-content"
          >
            <div
              className="user-list"
            />
          </div>,
          "className": "user-wrapper flex-container space-between",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user-list"
              />,
              false,
            ],
            "className": "user-detail-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "user-list",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            false,
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
